src/cast_core/src/CommitLatency.cpp

```diff
@@ -0,0 +1,195 @@
+// Copyright 2019-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cast_core/actors/CommitLatency.hpp>
+
+#include <memory>
+
+#include <yaml-cpp/yaml.h>
+
+#include <bsoncxx/json.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/database.hpp>
+
+#include <boost/log/trivial.hpp>
+
+#include <gennylib/Cast.hpp>
+#include <gennylib/ExecutionStrategy.hpp>
+#include <gennylib/InvalidConfigurationException.hpp>
+#include <gennylib/context.hpp>
+#include <value_generators/value_generators.hpp>
+
+
+namespace genny::actor {
+
+struct CommitLatency::PhaseConfig {
+    mongocxx::collection collection;
+    mongocxx::write_concern wc;
+    mongocxx::read_concern rc;
+    mongocxx::read_preference rp;
+    std::string rp_string;
+    std::shared_ptr<mongocxx::client_session> session;
+    mongocxx::options::find optionsFind;
+    mongocxx::options::update optionsUpdate;
+    mongocxx::options::aggregate optionsAggregate;
+    bool useSession;
+    int64_t repeat;
+    int64_t threads;
+    std::uniform_int_distribution<int64_t> amountDistribution;
+    ExecutionStrategy::RunOptions options;
+
+    PhaseConfig(PhaseContext& phaseContext, const mongocxx::database& db)
+        : collection{db[phaseContext.get<std::string>("Collection")]},
+          rp_string{phaseContext.get<std::string>("ReadPreference")},
+          useSession{phaseContext.get<bool, false>("Session")},
+          repeat{phaseContext.get<IntegerSpec>("Repeat")},
+          threads{phaseContext.get<IntegerSpec>("Threads")},
+          amountDistribution{-100, 100},
+          options{ExecutionStrategy::getOptionsFrom(phaseContext, "ExecutionsStrategy")} {
+              // write_concern
+              if( phaseContext.get<bool, false>("WriteConcernMajority") ) {
```

Comment:
do you want to allow `WriteConcernMajority` to specify the value in `milliseconds` (via a `TimeSpec`)?

```c++
if(auto maj = phaseContext.get<TimeSpec, false>("WriteConcernMajority") ) {
  wc.majorty(maj);
}
```

Reply:
Not really. I think a boolean is more intuitive and matches other options that are also boolean. But thanks for the example.